src/loops.rs

```diff
@@ -48,6 +48,16 @@ declare_lint!{ pub EXPLICIT_ITER_LOOP, Warn,
 declare_lint!{ pub ITER_NEXT_LOOP, Warn,
                "for-looping over `_.next()` which is probably not intended" }
 
+/// **What it does:** This lint checks for `for` loops over Option values. It is `Warn` by default.
+///
+/// **Why is this bad?** Readability. This is more clearly expressed as an `if let`.
```

Comment:
llvm will kill it. I don't think the performance after optimizations will be affected.